File: api/src/models/School.ts
import { Schema, model, Document } from 'mongoose';

export interface SchoolDocument extends Document {
  name: string;
  nemisCode: string;
  type: string;
  level: string;
  educationLevels: string[];
  location: {
    county: string;
    subCounty: string;
    ward: string;
    address?: string;
  };
  contactInfo: {
    phone?: string;
    email?: string;
    website?: string;
  };
  verified: boolean;
  createdAt: Date;
  updatedAt: Date;
}

const schoolSchema = new Schema<SchoolDocument>({
  name: { type: String, required: true },
  nemisCode: { type: String, required: true, unique: true },
  type: { type: String, required: true },
  level: { type: String, required: true },
  educationLevels: [{ type: String, required: true }],
  location: {
    county: { type: String, required: true },
    subCounty: { type: String, required: true },
    ward: { type: String, required: true },
    address: String
  },
  contactInfo: {
    phone: String,
    email: String,
    website: String
  },
  verified: { type: Boolean, default: false }
}, {
  timestamps: true
});

export const School = model<SchoolDocument>('School', schoolSchema);

export default School;

File: modules/shared-domain/src/main/java/ke/elimuconnect/domain/School.java
package ke.elimuconnect.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class School {
    private String id;
    private String name;
    private String nemisCode;
    private String type;
    private String level;
    private String county;
    private String subCounty;
    private String ward;
    private String location;
    private String phoneNumber;
    private String email;
    private String principalName;
    private String principalContact;
    private boolean active;
    private String createdAt;
    private String updatedAt;
    private String createdBy;
    private String updatedBy;
}
